### PR TITLE
apimachinery annual report for 2020

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/kubernetes/community/issues/5494

This provides the standard annual report.  Overall, we think api-machinery is in a healthy spot.

/cc @fedebongio 
/assign @dims

### CODE REVIEW

**Comment by fedebongio** [comment]
Thank you @deads2k !

File: sig-api-machinery/annual-report-2020.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+   don't know much about: this happens often when Kubernetes is upgrading libraries for example. 
+
+
+5. What metrics/community health stats does your group care about and/or measure? Examples?
+
+   On the technical health of the SIG, we look at
+   - the ratio of open/close PRs
+   - the ratio of open/close Issues
+   - overall age of open Issues
+   - Nnmber of active contributors to the sig
```
Comment (adamzhoul) [inline sig-api-machinery/annual-report-2020.md:149]: Nnmber -> Number

File: sig-api-machinery/annual-report-2020.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+## Current initiatives and project health
+1. What are initiatives that should be highlighted, lauded, shout outs, that your group is proud of? Currently underway?
+   What are some of the longer tail projects that your group is working on?
+   
+   Currently underway:
+   1. [server-side-apply](https://github.com/kubernetes/enhancements/issues/555) to GA
+   2. [server-side-apply client](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/2144-clientgo-apply#alternative-1-generated-structs-where-all-fields-are-pointers)
+   3. [optionally skip backend TLS verifiction](https://github.com/kubernetes/enhancements/issues/1295)
+   4. [namespace labels](https://github.com/kubernetes/enhancements/pull/2162)
+   5. Getting ready for CRD and admission webhook v1beta1 APIs: [reminder on kubernetes-dev](https://groups.google.com/g/kubernetes-dev/c/z_AE1EHhZF4/m/kBd3HkWxAwAJ).
```
Comment (sbueringer) [inline sig-api-machinery/annual-report-2020.md:96]:
```suggestion
   5. Getting ready for CRD and admission webhook v1beta1 APIs removal: [reminder on kubernetes-dev](https://groups.google.com/g/kubernetes-dev/c/z_AE1EHhZF4/m/kBd3HkWxAwAJ).
```
If you want to

File: sig-api-machinery/annual-report-2020.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+      Yes, the [README](https://github.com/kubernetes/community/blob/master/sig-api-machinery/README.md) is accurate.
+   2. All subprojects correctly mapped and listed in sigs.yaml?
+
+      Yes, our [subprojects](https://github.com/kubernetes/community/blob/master/sig-api-machinery/README.md#subprojects) are current.
+   3. What’s your meeting culture? Large/small, active/quiet, learnings? Meeting notes up to date?
+ 
+      Are you keeping recordings up to date/trends in community members watching recordings?
+   
+      We have two main meetings, both fairly small, with [notes and agenda up to date](https://docs.google.com/document/d/1x9RNaaysyO0gXHIr1y50QFbiL1x8OWnk2v3XnrdkT5Y/edit).
```
Comment (parispittman) [inline sig-api-machinery/annual-report-2020.md:35]: do you still do bug triage or did that fall back into main?

File: sig-api-machinery/annual-report-2020.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      [Our recordings](https://www.youtube.com/playlist?list=PL69nYSiGNLP21oW3hbLyjjj4XhrwKxH2R) are usually uploaded within two weeks.
+2. How does the group get updates, reports, or feedback from subprojects? 
+   Are there any springing up or being retired? Are OWNERS.md files up to date in these areas?
+   
+   We get updates on an ad-hoc basis.
+   We have approved some prototyping projects and have no plans to retire any at this time.
```
Comment (parispittman) [inline sig-api-machinery/annual-report-2020.md:41]: links to prototyping projects that you can share?

File: sig-api-machinery/annual-report-2020.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+   We have not actively pruned OWNERS, some people have been added to various subprojects.
+
+3. Same question as above but for working groups.
```
Comment (parispittman) [inline sig-api-machinery/annual-report-2020.md:44]: do you still sponsor component standard and mulitenancy?

File: sig-api-machinery/annual-report-2020.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+   We don’t measure membership.
+
+3. How does the group measure reviewer and approver bandwidth?
+   Do you need help in any area now? What are you doing about it?
+   
+   Our predicted rate of feature delivery and stability roughly matches the achieved rate.
+   While we would be happy to see developers move up the ladder, we don’t see a pressing need to adjust the current rate.
+   
+   We perform twice a week triage and our [issue open/close rates are holding steady](https://k8s.devstats.cncf.io/d/39/issues-opened-closed-by-sig?orgId=1&var-period=d7&var-sig_name=api-machinery&var-kind_name=All).
```
Comment (parispittman) [inline sig-api-machinery/annual-report-2020.md:66]: ty for this link

File: sig-api-machinery/annual-report-2020.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+   We don’t participate in any particular programs.
+   We find many contributors via slack, PRs, and issues.
+
+6. Does the group have contributors from multiple companies/affiliations?
+   Can end users/companies contribute in some way that they currently are not?
+   
+   Yes, there are contributors from [multiple companies](https://k8s.devstats.cncf.io/d/74/contributions-chart?orgId=1&var-period=d7&var-metric=contributions&var-repogroup_name=SIG%20API%20Machinery&var-country_name=All&var-company_name=All&var-company=all).
+   We see all sorts of contributions, varying from issues, to comments, to PRs, to designs, to sig meeting participation,
+   and user-survey data.
```
Comment (parispittman) [inline sig-api-machinery/annual-report-2020.md:84]: could end users support you more in some way?

File: sig-api-machinery/annual-report-2020.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+
+## Current initiatives and project health
+1. What are initiatives that should be highlighted, lauded, shout outs, that your group is proud of? Currently underway?
```
Comment (parispittman) [inline sig-api-machinery/annual-report-2020.md:88]: anything that y'all are super proud of from 2020?

File: sig-api-machinery/annual-report-2020.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+   We are working on mitigating the impact of removing beta APIs in 1.22.
+
+4. What areas and/or subprojects does the group need the most help with?
+
+   The SIG has 2 Working Groups that are largely independent. 
+   
+   There are several areas where regularly the SIG becomes under pressure, especially closer to code freezes and the
+   vast amount of code owned by API Machinery.
+   
+   The ecosystem of the different Kubernetes Clients that we own grows more or less organically. Client-go and
```
Comment (parispittman) [inline sig-api-machinery/annual-report-2020.md:136]: do you think we should do more outreach in those communities (python, go, etc) for any reason?

**Comment by dims** [comment]
@deads2k @fedebongio is this ready to merge today? thanks!

File: sig-api-machinery/annual-report-2020.md (hunk 9 above)
Comment (deads2k) [reply sig-api-machinery/annual-report-2020.md:136]: > do you think we should do more outreach in those communities (python, go, etc) for any reason?

I had not really considered the question.  Does this doc need to reflect future objectives or merely an honest assessment of the current state?

File: sig-api-machinery/annual-report-2020.md (hunk 3 above)
Comment (fedebongio) [reply sig-api-machinery/annual-report-2020.md:35]: We still do it, twice a week, no excuses :)

**Pull request merged**